src/mpid/ch4/src/ch4r_callbacks.h

```diff
@@ -634,4 +634,43 @@ static inline int MPIDI_send_long_ack_target_msg_cb(int handler_id, void *am_hdr
     goto fn_exit;
 }
 
+#undef FUNCNAME
+#define FUNCNAME MPIDI_comm_abort_origin_cb
+#undef FCNAME
+#define FCNAME MPL_QUOTE(FUNCNAME)
+static inline int MPIDI_comm_abort_origin_cb(MPIR_Request * sreq)
+{
+    int mpi_errno = MPI_SUCCESS;
+    MPIR_FUNC_VERBOSE_STATE_DECL(MPID_STATE_MPIDI_COMM_ABORT_ORIGIN_CB);
+    MPIR_FUNC_VERBOSE_ENTER(MPID_STATE_MPIDI_COMM_ABORT_ORIGIN_CB);
+    MPID_Request_complete(sreq);
+    MPIR_FUNC_VERBOSE_EXIT(MPID_STATE_MPIDI_COMM_ABORT_ORIGIN_CB);
+    return mpi_errno;
+}
+
+#undef FUNCNAME
+#define FUNCNAME MPIDI_comm_abort_target_msg_cb
+#undef FCNAME
+#define FCNAME MPL_QUOTE(FUNCNAME)
+static inline int MPIDI_comm_abort_target_msg_cb(int handler_id, void *am_hdr,
```

Comment:
We should make this a `void` function. It will never return.

Reply:
That might be a problem because of the registration function in `generic/mpidig_init.c` takes `MPIDIG_am_target_msg_cb` type parameter which explicitly says the return value be `int`.

Reply:
On a second thought, let me try it first. Since return type is not part of the function signature, it might work.

Reply:
Ah, ok. Well let's see how it goes. We should do whatever makes the compiler happy in that case.

Reply:
The code is running fine. But compiler gives a warning of incompatible pointer type for that. I would leave it as `int` just for the cleanness.